### PR TITLE
Updated psmeca to prevent double-correcting for map rotation

### LINKED ISSUE #6528: psmeca incorrectly rotates nodal planes of some mechanisms in oblique projections
**Description of the problem**

psmeca.c calculates a rotation angle called delaz that is added to nodal plane strikes to correct for map rotation in oblique projections. In some edge cases (mainly for pure double couple mechanisms), psmeca recalculates nodal plane strikes from principal axes. Because the axes are corrected by delaz already, a subsequent correction to nodal plane strike causes over-rotation of the nodal planes. The proposed code fix in the pull request for this issue simply checks whether a correction has already been applied before the second correction is done.

### PR DESCRIPTION
**Pull request by kyleedwardbradley**
**Description of proposed changes**

Add a check to psmeca.c to avoid double-correction of some nodal plane strikes in oblique projections

Fixes #6528

**Reminders**

- [ ] Make sure that your code follows our style. Use the other functions/files as a basis.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Describe changes to function behavior and arguments in a comment below the function declaration.
- [ ] If adding new functionality, add a detailed description to the documentation and/or an example.

### CODE REVIEW

**Review by meghanrjones — APPROVED** [review]
Looks good to me, thanks!

FYI I'll ping another maintainer who has worked a lot more than me on meca before merging. All the tests pass and the forum example works well.

**Review by seisman — APPROVED** [review]

**Pull request merged**
